core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionWitness.scala

```diff
@@ -69,7 +89,8 @@ object TransactionWitness {
     if (witnesses.exists(_ != EmptyScriptWitness)) {
       TransactionWitnessImpl(witnesses)
     } else {
-      EmptyWitness
+      //means that everything must be a empty ScriptWitness
+      EmptyWitness(witnesses.map(_.asInstanceOf[EmptyScriptWitness.type]))
```

Comment:
nit: no strong feelings about this but you could also just do `EmptyWitness.fromN(witnesses.length)` to avoid casting here